packages/select/src/index.mts

```diff
@@ -11,113 +11,101 @@ import {
   isNumberKey,
   Separator,
   AsyncPromptConfig,
+  type Layout,
+  index,
 } from '@inquirer/core';
 import type {} from '@inquirer/type';
 import chalk from 'chalk';
 import figures from 'figures';
 import ansiEscapes from 'ansi-escapes';
 
-type Choice<Value> = {
+export type Choice<Value> = {
   value: Value;
   name?: string;
   description?: string;
   disabled?: boolean | string;
   type?: never;
 };
 
+type Item<Value> = Separator | Choice<Value>;
+
+const selectable = <Value,>(item: Item<Value>): item is Choice<Value> =>
+  !Separator.isSeparator(item) && !item.disabled;
+
+const render = <Value,>({ item, active }: Layout<Item<Value>>) => {
```

Comment:
Similar to `index`, I'd prefer to not expose extra utility types from `@inquirer/core`. Would types be inferred properly without typing it?

Otherwise, let's just define locally `item` & `active` at the cost of cuplication.